src/Kobo/PowerOff.cpp

```diff
@@ -84,10 +85,13 @@ DrawBanner(Canvas &canvas, PixelRect &rc)
   canvas.DrawText({x, rc.top + int(banner_height - normal_font.GetHeight()) / 2},
                   website);
 
-  const TCHAR *const comment = _T("powered off");
+  TCHAR comment[30] = _T("powered off");
+  TCHAR battery[4];
+  if (File::ReadString(Path("/sys/class/power_supply/mc13892_bat/capacity"), battery, sizeof(battery))){
+          snprintf ( comment+strlen(comment), 30, " - battery %s%%", battery);
```

Comment:
This is still a stack buffer overflow, because you pretend the buffer is 30 bytes **plus** the current length of the comment.